src/pyfaf/actions/repomod.py

```diff
@@ -29,13 +31,38 @@ def __init__(self):
         self.repo_types = pyfaf.repos.repo_types
 
     def run(self, cmdline, db):
-        repo = (db.session.query(Repo)
-                .filter(Repo.name == cmdline.REPO)
-                .first())
+        if not cmdline.REPO and not cmdline.all:
+            self.log_error("No repositories specified")
+            return 1
+
+        repos = []
+
+        for pattern in cmdline.REPO:
+            wildcard_used = any(c in pattern for c in ['?', '*'])
+            if wildcard_used:
+                break
+
+        if (len(cmdline.REPO) > 1 or cmdline.all or wildcard_used) and (cmdline.name or cmdline.nice_name):
+            self.log_error("Can't assign the same name to multiple repositories")
+            return 1
+
 
-        if not repo:
-            self.log_error("Repository '{0}' not found"
-                           .format(cmdline.REPO))
+        if cmdline.all or '*' in cmdline.REPO:
+            repos.extend(db.session.query(Repo).all())
+
+        else:
+            for pattern in cmdline.REPO:
+                pattern = re.sub('_', r'\\_', pattern)
+                pattern = re.sub('%', r'\\%', pattern)
+                pattern = re.sub(r'\*', '%', pattern)
+                pattern = re.sub(r'\?', '_', pattern)
+                repos.extend(db.session.query(Repo).filter(Repo.name.like(pattern)).all())
```

Comment:
Can you make it a function (same code is also in `repodel`) ?

Reply:
Done.